Danki2/Assets/Scripts/Abilities/Ability Tree/Node.cs

```diff
@@ -31,9 +37,19 @@ public Node GetChild(Direction direction)
         return _children[direction];
     }
 
-    public void SetChild(Direction direction, Node value)
+    public void SetChild(Direction direction, Node child)
+    {
+        _children[direction] = child;
+        child.Parent = this;
+
+        Subscription subscription = childChangeSubscriptions[direction];
+        if (subscription != null) subscription.Unsubscribe();
+        subscription = child.ChangeSubject.Subscribe(ChangeSubject.Next);
```

Comment:
I think getting the subscription from the dictionary and setting it is a bit confusing, it's not intuitive that this is updating the value in the dictionary. How about:
```suggestion
        childChangeSubscriptions[direction]?.Unsubscribe();
        childChangeSubscriptions[direction] = child.ChangeSubject.Subscribe(ChangeSubject.Next);
```